packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -48,9 +48,13 @@ const StoreLocatorModal = ({isOpen, onClose = noop}) => {
             maxDistance: STORE_LOCATOR_DISTANCE
         }
     })
-    var storesInfo = []
-    if (searchStoresData.data !== undefined && searchStoresData.data.data !== undefined)
-        storesInfo = searchStoresData.data.data
+
+    const storesInfo =
+        searchStoresData.data !== undefined
+            ? searchStoresData.data.data !== undefined
+                ? searchStoresData.data.data
+                : []
+            : undefined
```

Comment:
I think we can make it clearer why there are 3 possible scenarios. Since react-query gives us a way to find out whether it's still loading/fetching the data, we can do something like this:

```js
    var {data: searchStoresData, isLoading} = useSearchStores({parameters})

    const storesInfo = isLoading ? undefined : searchStoresData.data || []

```